client/index.js

```diff
@@ -47,7 +47,11 @@ function log(level, msg) {
 
 // Send messages to the outside, so plugins can consume it.
 function sendMsg(type, data) {
-	if(typeof self !== "undefined" && self.window) {
+	if(
+		typeof self !== "undefined" &&
+		(typeof WorkerGlobalScope === "undefined" ||
+		!(self instanceof self.WorkerGlobalScope))
```

Comment:
I don't understand why you're using `self.WorkerGlobalScope` here. `self instanceof WorkerGlobalScope` should be correct.